.gitlab-ci.yml

```diff
@@ -8,67 +8,67 @@ stages:
   - prod
 
 variables:                         &default-vars
-  # DOCKER_HOST:                     tcp://docker:2375
-  DOCKER_HOST:                     tcp://localhost:2375
-  DOCKER_DRIVER:                   overlay2
   IMAGE_TAG:                       latest
-  REGISTRY_PATH:                   paritytech
+  REGISTRY_PATH:                   docker.io/paritytech
 
 default:
   cache:                           {}
 
 .build:                            &docker_build
   stage:                           build
-  image:                           docker:stable
-  services:
-    - docker:dind
+  image:                           quay.io/buildah/stable
   rules:
     - if: $IMAGE_NAME == $CI_JOB_NAME
   tags:
     - kubernetes-parity-build
 
-# Push to Dockerhub
+# Push to Dockerhub using buildah
 .push_to_docker_hub:               &push_to_docker_hub
   - export IMAGE_DATE_TAG="$CI_COMMIT_SHORT_SHA-$(date +%Y%m%d)"
-  - docker build --no-cache
+  - buildah version
+  - buildah bud
+      --squash
+      --format=docker
       --build-arg VCS_REF="$CI_COMMIT_SHA"
       --build-arg BUILD_DATE="$(date +%Y%m%d)"
-      --build-arg REGISTRY_PATH=$REGISTRY_PATH
-      --tag $REGISTRY_PATH/$IMAGE_NAME:$IMAGE_DATE_TAG
-      --tag $REGISTRY_PATH/$IMAGE_NAME:$IMAGE_TAG
-      --file dockerfiles/$IMAGE_NAME/Dockerfile dockerfiles
-  - docker login -u "$DOCKER_USER" -p "$DOCKER_PASSWORD"
-  - docker info
-  - docker push $REGISTRY_PATH/$IMAGE_NAME:$IMAGE_DATE_TAG
-  - docker push $REGISTRY_PATH/$IMAGE_NAME:$IMAGE_TAG
-  - docker logout
+      --build-arg REGISTRY_PATH="$REGISTRY_PATH"
+      --tag "$REGISTRY_PATH/$IMAGE_NAME:$IMAGE_DATE_TAG"
+      --tag "$REGISTRY_PATH/$IMAGE_NAME:$IMAGE_TAG"
+      --file "dockerfiles/$IMAGE_NAME/Dockerfile" dockerfiles
+  - buildah info
+  - buildah login --username "$DOCKER_USER" --password "$DOCKER_PASSWORD" "$REGISTRY_PATH"
```

Comment:
At least for docker, it's recommended that the password is read from stdin rather than using the --password flag for safety reasons ("Using STDIN prevents the password from ending up in the shell’s history, or log-files."). in the buildah docs they do this like this:

`echo $testpassword | buildah login -u testuser --password-stdin docker.io`

Reply:
cheers!